src/screenshot/sentry_screenshot_windows.c

```diff
@@ -0,0 +1,152 @@
+        = (GdiplusStartup_t)GetProcAddress(gdiplus, "GdiplusStartup");
+    GdipCreateBitmapFromHBITMAP_t pGdipCreateBitmapFromHBITMAP
+        = (GdipCreateBitmapFromHBITMAP_t)GetProcAddress(
+            gdiplus, "GdipCreateBitmapFromHBITMAP");
+    GdipSaveImageToFile_t pGdipSaveImageToFile
+        = (GdipSaveImageToFile_t)GetProcAddress(gdiplus, "GdipSaveImageToFile");
+    GdipDisposeImage_t pGdipDisposeImage
+        = (GdipDisposeImage_t)GetProcAddress(gdiplus, "GdipDisposeImage");
+
+    if (!pGdiplusStartup || !pGdipCreateBitmapFromHBITMAP
+        || !pGdipSaveImageToFile || !pGdipDisposeImage) {
+        FreeLibrary(gdiplus);
+        return false;
+    }
+
+    GdiplusStartupInput input;
+    ZeroMemory(&input, sizeof(input));
+    input.GdiplusVersion = 1;
+
+    ULONG_PTR token = 0;
+    if (pGdiplusStartup(&token, &input, NULL) != GpOk) {
+        FreeLibrary(gdiplus);
+        return false;
+    }
+
+    GpImage *image = NULL;
+    if (pGdipCreateBitmapFromHBITMAP(bitmap, NULL, &image) != GpOk) {
+        FreeLibrary(gdiplus);
+        return false;
+    }
+
+    // CLSIDFromString(L"{557cf406-1a04-11d3-9a73-0000f81ef32e}")
+    CLSID GdiPngEncoder = { 0x557cf406, 0x1a04, 0x11d3,
+        { 0x9a, 0x73, 0x00, 0x00, 0xf8, 0x1e, 0xf3, 0x2e } };
+    GpStatus rv = pGdipSaveImageToFile(image, path, &GdiPngEncoder, NULL);
+    pGdipDisposeImage(image);
+    FreeLibrary(gdiplus);
+    return rv == GpOk;
+}
+
+static void
+calculate_region(DWORD pid, HRGN region)
+{
+    HMODULE dwmapi = LoadLibraryW(L"dwmapi.dll");
+    if (!dwmapi) {
+        return;
+    }
+    DwmGetWindowAttribute_t pDwmGetWindowAttribute
+        = (DwmGetWindowAttribute_t)GetProcAddress(
+            dwmapi, "DwmGetWindowAttribute");
+    if (!pDwmGetWindowAttribute) {
+        return;
+    }
+
+    HWND hwnd = GetShellWindow();
+    while (hwnd) {
+        if (IsWindowVisible(hwnd)) {
+            RECT frame = { 0 };
+            pDwmGetWindowAttribute(
+                hwnd, DWMWA_EXTENDED_FRAME_BOUNDS, &frame, sizeof(RECT));
+            if (frame.right > frame.left && frame.bottom > frame.top) {
+                DWORD wpid = 0;
+                GetWindowThreadProcessId(hwnd, &wpid);
+                HRGN wregion = CreateRectRgnIndirect(&frame);
+                if (wpid == pid) {
+                    CombineRgn(region, region, wregion, RGN_OR);
+                } else {
+                    CombineRgn(region, region, wregion, RGN_DIFF);
+                }
+                DeleteObject(wregion);
+            }
+        }
+        hwnd = GetWindow(hwnd, GW_HWNDPREV);
+    }
+}
+
+bool
+sentry__screenshot_capture(const sentry_path_t *path)
+{
+    HRGN region = CreateRectRgn(0, 0, 0, 0);
+    calculate_region(GetCurrentProcessId(), region);
+
+    RECT box;
+    GetRgnBox(region, &box);
+    LONG width = box.right - box.left;
+    LONG height = box.bottom - box.top;
+    if (width <= 0 || height <= 0) {
+        DeleteObject(region);
```

Comment:
The main reason for this is that none of the Windows were visible at the time of the crash. Let's log this as well, even though most people probably won't have the logs active in production, that seems to change.